### PR TITLE
otelcol.auth.headers: new component

### LINKED ISSUE #2356: Flow: otelcol.auth.headers component

### PR DESCRIPTION
**Pull request by rfratto**
Introduce an `otelcol.auth.headers` component which wraps around the upstream `headerssetter` extension and allows other components to use it.

Closes #2356.

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.auth.headers.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    value = "HEADER_VALUE"
+  }
+  token = "TOKEN"
+}
+```
+
+## Arguments
+
+`otelcol.auth.headers` doesn't support any arugments and is configured fully
+through inner blocks.
```
Comment (karengermond) [inline docs/sources/flow/reference/components/otelcol.auth.headers.md:34]:
```suggestion
`otelcol.auth.headers` doesn't support any arguments and is configured fully
through inner blocks.
```

**Review by karengermond — APPROVED** [review]
LGTM -- one little issue

File: docs/sources/flow/reference/components/otelcol.auth.headers.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+configuration.
+
+## Debug information
+
+`otelcol.auth.headers` does not expose any component-specific debug information.
+
+## Example
+
+This example configures [otelcol.exporter.otlp][] to use headers token
+authentication:
```
Comment (rfratto) [inline docs/sources/flow/reference/components/otelcol.auth.headers.md:93]:
```suggestion
This example configures [otelcol.exporter.otlp][] to use custom headers:
```

File: docs/sources/flow/reference/components/otelcol.auth.headers.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+## Usage
+
+```river
+otelcol.auth.headers "LABEL" {
+  header {
+    key   = "HEADER_NAME"
+    value = "HEADER_VALUE"
+  }
+  token = "TOKEN"
```
Comment (tpaschalis) [inline docs/sources/flow/reference/components/otelcol.auth.headers.md:27]:
```suggestion

```

**Review by tpaschalis — APPROVED** [review]

File: docs/sources/flow/reference/components/otelcol.auth.headers.md (hunk 3 above)
Comment (rfratto) [reply docs/sources/flow/reference/components/otelcol.auth.headers.md:27]: doh, thanks :)

File: component/otelcol/auth/headers/headers.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+func (h *Header) UnmarshalRiver(f func(interface{}) error) error {
+	type header Header
+	if err := f((*header)(h)); err != nil {
+		return err
+	}
+
+	switch {
+	case h.Key == "":
+		return fmt.Errorf("key must be set to a non-empty string")
+	case h.FromContext == nil && h.Value == nil:
```
Comment (mattdurham) [inline component/otelcol/auth/headers/headers.go:90]: Should we also check for empty strings?

Comment (rfratto) [reply component/otelcol/auth/headers/headers.go:90]: I copied this from upstream. It's valid for a header value to be empty, but not the header key, which is why I think they don't check for empty strings here

**Pull request merged**